packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -358,6 +366,31 @@ abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget>
   /// [forEachTween] should override [didUpdateTweens] to update those
   /// properties. Dependent properties should not be updated within
   /// [forEachTween].
+  ///
+  /// {@tool sample}
+  /// ```dart
+  /// class MyWidgetState extends AnimatedWidgetBaseState<MyWidget> {
+  ///   ColorTween _colorTween;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return Text(
+  ///       'Hello World',
+  ///       style: TextStyle(color: _colorTween.evaluate(animation)),
+  ///     );
+  ///   }
+  ///
+  ///   @override
+  ///   void forEachTween(TweenVisitor<dynamic> visitor) {
+  ///     _colorTween = visitor(
```

Comment:
Can you offer a brief comment in the sample code that illuminates the purpose of each line here?  I find the visitor system within implicit animations to appear very perplexing and I always seem to forget what it is that I'm trying to accomplish when I write `forEachTween()`.

Reply:
Will do.